### PR TITLE
[Cppcheck] Provide Sider's recommended set of rules

### PR DESCRIPTION
**Pull request by sleeeto**
> Explain a summary, purpose, or background for this change.

Provide Sider's recommended set of rules for Cppcheck.

> Refer related issues or pull requests (e.g. `Close #123` or `None`).

https://github.com/sider/sideci/issues/10671

> Check the following if needed.

- [x] Add a new entry to [CHANGELOG.md](https://github.com/sider/runners/blob/master/CHANGELOG.md) if this change is notable.

### CODE REVIEW

File: images/cppcheck/sider_recommended_cppcheck.txt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+toomanyconfigs
```
Comment (suetsuguk) [inline images/cppcheck/sider_recommended_cppcheck.txt:1]: [suggestion] Arrange items in alphabetical order.

Comment (sleeeto) [reply images/cppcheck/sider_recommended_cppcheck.txt]: Thanks, I fixed it.

**Review by suetsuguk — APPROVED** [review]
LGTM. Thank you.

File: CHANGELOG.md [hunk 2]
```diff
@@ -39,6 +39,8 @@ Misc:
 - Drop support for JCenter repository [#2429](https://github.com/sider/runners/pull/2429)
 - **JSHint** Enhance JSHint runner [#2430](https://github.com/sider/runners/pull/2430)
 - **PMD CPD** Fix supported languages [#2432](https://github.com/sider/runners/pull/2432)
+- **PHPMD** Replace XML reporter with JSON one [#2435](https://github.com/sider/runners/pull/2435)
+- **Cppcheck** Provide new recommended configuration [#2443](https://github.com/sider/runners/pull/2443)
```
Comment (ybiquitous) [inline CHANGELOG.md:43]: Why 2 lines change?

Comment (sleeeto) [reply CHANGELOG.md]: I missed it, and fixed it.

**Pull request merged**